### PR TITLE
NodeHash in Proof struct

### PR DESCRIPTION
**Pull request by dirtydanee**
- Adding support for executing multiple async operations 
- Refactor the `proof` function on `ChainpointService` using operations
- Testing like crazy

### CODE REVIEW

File: BaaServiceKitTests/Mocks/Error+BaaServiceKitTests.swift [hunk 1]
```diff
@@ -0,0 +1,3 @@
+enum E: Swift.Error {
```
Comment (szuniverse) [inline BaaServiceKitTests/Mocks/Error+BaaServiceKitTests.swift]: I think `E`enum name is not so talkative. Should use `MockingError` or `TestingError` or something like this.

File: BaaServiceKitTests/Mocks/MockAPIClient.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+//
+
+import Foundation
+@testable import BaaServiceKit
+
+final class MockAPIClient: APIClient {
+
+    var result: Result<APIResponse>?
+
+    override func execute(request: BlockchainRequest, completion: @escaping (Result<APIResponse>) -> Swift.Void) {
```
Comment (szuniverse) [inline BaaServiceKitTests/Mocks/MockAPIClient.swift:16]: Why do you use `Swift.Void`? Why dont enough `Void`?

File: BaaServiceKitTests/Resources/ChainpontResponses/FakeProofResponse1.json [hunk 3]
```diff
@@ -1,9 +0,0 @@
-[
```
Comment (szuniverse) [inline BaaServiceKitTests/Resources/ChainpontResponses/FakeProofResponse1.json:1]: Why did you delete this?

File: BaaServiceKitTests/Resources/ChainpontResponses/FakeProofResponse2.json [hunk 4]
```diff
@@ -1,13 +0,0 @@
-// Examples
-// 1. None
-//[
```
Comment (szuniverse) [inline BaaServiceKitTests/Resources/ChainpontResponses/FakeProofResponse2.json:3]: Why did you delete this?

File: Source/BlockchainService/AsynchronousOperation.swift [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+            case .executing:
+                return \.isExecuting
+            case .finished:
+                return \.isFinished
+            }
+        }
+    }
+}
+
+private extension AsynchronousOperation {
```
Comment (szuniverse) [inline Source/BlockchainService/AsynchronousOperation.swift:73]: line:53 there is a private extension, isn't it possible to merge these?

File: BaaServiceKitTests/Mocks/Error+BaaServiceKitTests.swift (hunk 1 above)
Comment (dirtydanee) [reply BaaServiceKitTests/Mocks/Error+BaaServiceKitTests.swift]: `MockingError` sounds nice

File: BaaServiceKitTests/Mocks/MockAPIClient.swift (hunk 2 above)
Comment (dirtydanee) [reply BaaServiceKitTests/Mocks/MockAPIClient.swift:16]: Its just coming from SwiftLint, I prefer to use `Swift.Void`

File: BaaServiceKitTests/Resources/ChainpontResponses/FakeProofResponse1.json (hunk 3 above)
Comment (dirtydanee) [reply BaaServiceKitTests/Resources/ChainpontResponses/FakeProofResponse1.json:1]: it was not used. should I keep it?

File: BaaServiceKitTests/Resources/ChainpontResponses/FakeProofResponse2.json (hunk 4 above)
Comment (dirtydanee) [reply BaaServiceKitTests/Resources/ChainpontResponses/FakeProofResponse2.json:3]: it was not used. should I keep it? The swagger should have it anyway.

File: Source/BlockchainService/AsynchronousOperation.swift (hunk 5 above)
Comment (dirtydanee) [reply Source/BlockchainService/AsynchronousOperation.swift:73]: One is `private` functions, the other is a `type` declaration.
